tme/handlers.go

```diff
@@ -0,0 +1,193 @@
+package tme
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"strings"
+
+	"github.com/Financial-Times/go-fthealth/v1a"
+	"github.com/Financial-Times/service-status-go/gtg"
+	"github.com/Financial-Times/transactionid-utils-go"
+	log "github.com/Sirupsen/logrus"
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+type Handler struct {
+	service Service
+}
+
+func NewHandler(service Service) *Handler {
+	return &Handler{
+		service: service,
+	}
+}
+
+func (th *Handler) HandleGetFullTaxonomy(resp http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	t := vars["type"]
+
+	pv, err := th.service.GetAllConcepts(t)
+
+	if err != nil {
+		writeJSONMessageWithStatus(resp, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer pv.Close()
+	resp.WriteHeader(http.StatusOK)
+	io.Copy(resp, &pv)
+}
+
+func (th *Handler) HandleGetSingleConcept(resp http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	t := vars["type"]
+	uuid := vars["uuid"]
+
+	resp.Header().Add("Content-Type", "application/json")
+
+	obj, found, err := th.service.GetConceptByUUID(t, uuid)
+	if err != nil {
+		writeJSONMessageWithStatus(resp, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	writeJSONResponse(obj, found, EndpointTypeMappings[t]["type"].(string), resp)
+}
+
+func (th *Handler) GetIDs(resp http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	t := vars["type"]
+
+	pv, err := th.service.GetConceptUUIDs(t)
+
+	if err != nil {
+		writeJSONMessageWithStatus(resp, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer pv.Close()
+	resp.WriteHeader(http.StatusOK)
+	io.Copy(resp, &pv)
+}
+
+func (th *Handler) GetCount(resp http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	t := vars["type"]
+
+	count, err := th.service.GetCount(t)
+	if err != nil {
+		resp.Header().Add("Content-Type", "application/json")
+		writeJSONMessageWithStatus(resp, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	resp.Write([]byte(strconv.Itoa(count)))
+}
+
+func (th *Handler) HandleSendConcepts(resp http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	t := vars["type"]
+
+	jobID := strings.Replace(transactionidutils.NewTransactionID(), "tid", "job", -1)
+
+	go func(th *Handler, t string, jobID string) {
+		th.service.SendConcepts(t, jobID)
+	}(th, t, jobID)
+
+	resp.WriteHeader(http.StatusAccepted)
+	writeJSONResponse(map[string]interface{}{
+		"jobID": jobID,
+	}, true, t, resp)
+
+}
+
+func (th *Handler) HandleReloadConcepts(resp http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	t := vars["type"]
+	go th.service.Reload(t)
+
+	writeJSONMessageWithStatus(resp, fmt.Sprintf("Reloading %s", t), http.StatusAccepted)
+}
+
+func writeJSONMessageWithStatus(w http.ResponseWriter, msg string, statusCode int) {
+	w.WriteHeader(statusCode)
+	fmt.Fprintln(w, fmt.Sprintf("{\"message\": \"%s\"}", msg))
```

Comment:
Leave it. There doesn't seem to be any agreement on the standard content type for json stream